src/components/Header/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { Stack, Typography } from "@mui/material"
+import { useAuthState } from "react-firebase-hooks/auth";
+import { ChildType } from "../../types"
+import AddIcon from '@mui/icons-material/Add';
+import Button from "../Button"
+import { HeaderWrapper, LogoImg, LogOutButton, Block } from "./styles"
+import { auth } from "../..";
+import { signOut } from "@firebase/auth";
+import { FlexboxProps, StackBaseProps } from "@mui/system";
+
+type HeaderProps = {
+  children?: ChildType
+}
+
+type JustifyContentProperty = 'space-between' | 'center' | 'flex-start' | 'flex-end'
+
+type RowProps = StackBaseProps & {
+  children: ChildType,
+  padding?: number | string,
+  justify?: JustifyContentProperty,
+}
+
+const Row = ({ padding = 0, justify = 'space-between', ...rest }: RowProps) => <Stack direction="row" justifyContent={justify} alignItems="center" sx={{
+  padding: `${typeof padding === 'number' ? `${padding}px` : padding}`
+}} {...rest} />
```

Comment:
so in this case we're using padding shorthand to specify all the sides in a single expression, like this: `<Row padding="0 0 0 18px">` 

So a number isn't going to always work, it could be an array of numbers but this doesn't especially reduce the complexity

Reply:
Ahh, I see, that makes alot of sense!